### PR TITLE
updated NN and DL

### LINKED ISSUE #15: Improve Neural Networks and Deep Learning section
### Deep learning section review (@ankurshukla03):
- unanswered questions/issues:
  - how does the network learn? (this looks like a good place to introduce supervised learning)
  - there's no mention anywhere of gradient descent/backpropagation

### PR DESCRIPTION
**Pull request by ankurshukla03**
Closes #15
Updated section for Neural Network and Deep Learning. Made a different section for Neural network.

### CODE REVIEW

**Comment by tristaaan** [comment]
Something strange happened here, I lost a paragraph that I had written and was merged. 

https://github.com/diegocasmo/ml-project/pull/11/files#diff-7328e1b8dad8d695b78b0dc848c91223R41

If you see writing missing look at #11.

**Comment by diegocasmo** [comment]
@tristaaan 

Maybe this branch is not rebased with master?

**Comment by tristaaan** [comment]
Says it's up to date

**Review by tristaaan — APPROVED** [review]
I wonder if @diegocasmo should look over this to make sure none of his sections were zorched too

**Comment by diegocasmo** [comment]
> I wonder if @diegocasmo should look over this to make sure none of his sections were zorched too

I will, as soon as I finish watching Youtube, lol

**Review by diegocasmo — CHANGES_REQUESTED** [review]
- [x] Figures need to have citations (where were those found?)
- [x] The caption of the figures need to have a better explanation (what is being shown? what does ``x1``, ``w1`` mean?)
- [x] ``You can see figure \ref{fig:fig1} which shows the basic idea about how the neurons are connected``, maybe we can come up with a better phrasing for this? -> Figure \ref{fig:fig1} shows how a neural network receives a number of inputs, which are connected to an activation function through weighted edges [...]?
- [x] Inconsistent use of 'neural network' and 'NN'. Once the notation 'NN' is introduced, it should always be used instead of 'neural network'
- ``Sometimes there are no inputs this is where bias (one extra input) comes into play which has its own weight with an input of 1 just to make neuron activated even if there's no input`` -> I wasn't aware of this (though it makes sense if we look at the activation function). Where's this coming from (citation)?
- [x] Relu, ReLu, or ReLU? [Wikipedia article](https://en.wikipedia.org/wiki/Rectifier_(neural_networks)) favors ReLU
- I see no need to introduce the concept of RMSE in the NN section (we will talk about it in the experiment)
- [x] If I recall correctly, Olle suggested not talking in terms of 'We' in the report. There are a few places where that type of wording is used. If what I recall is true, then it should be avoided.
- [x] Since backpropagation formulas are not referenced anywhere, they shouldn't have a number besides them (I think `\nonumber` solves this, solved with `equation*`)
- [x] ``VGG-16 won the got second place to Google``?
- [x] There's some overlap between the first and last paragraphs of the VGG-16 section (talking about how we need to update the FCL size, number of classes it was originally trained for, etc)
- [x] Is it possible to have the caption of the VGG-16 table below (like the images)?

**Review by diegocasmo — APPROVED** [review]
Good work team!
